src/main/java/org/graylog/integrations/aws/transports/KinesisTransport.java

```diff
@@ -93,37 +92,63 @@ public void handleChangedThrottledState(boolean isThrottled) {
-        );
+        final String assumeRoleArn = configuration.getString(AWSInput.CK_ASSUME_ROLE_ARN);
+
+        final String dynamodbEndpoint = configuration.getString(AWSInput.CK_DYNAMODB_ENDPOINT);
+        final String cloudwatchEndpoint = configuration.getString(AWSInput.CK_CLOUDWATCH_ENDPOINT);
+        final String iamEndpoint = configuration.getString(AWSInput.CK_IAM_ENDPOINT);
+        final String kinesisEndpoint = configuration.getString(AWSInput.CK_KINESIS_ENDPOINT);
+
+        // Validate the endpoints, because the Kinesis client will silently fail if an endpoint is invalid.
+        validateEndpoint(dynamodbEndpoint, "DynamoDB");
+        validateEndpoint(cloudwatchEndpoint, "CloudWatch");
+        validateEndpoint(iamEndpoint, "IAM");
+        validateEndpoint(iamEndpoint, "Kinesis");
+
+        // Creating and passing an AWSRequest here is not ideal (since it's a web request object), but at least the
+        // builder can be used for all of the required request fields. Perhaps this can be improved later.
+        final AWSRequest awsRequest = AWSRequestImpl.builder()
+                                                    .region(region.id())
+                                                    .awsAccessKeyId(key)
+                                                    .awsSecretAccessKey(secret)
+                                                    .assumeRoleArn(assumeRoleArn)
+                                                    .cloudwatchEndpoint(cloudwatchEndpoint)
+                                                    .dynamodbEndpoint(dynamodbEndpoint)
+                                                    .iamEndpoint(iamEndpoint)
+                                                    .kinesisEndpoint(kinesisEndpoint).build();
+
+        final int batchSize = configuration.getInt(CK_KINESIS_RECORD_BATCH_SIZE, DEFAULT_BATCH_SIZE);
+        final String streamName = configuration.getString(CK_KINESIS_STREAM_NAME);
+        final AWSMessageType awsMessageType = AWSMessageType.valueOf(configuration.getString(AWSCodec.CK_AWS_MESSAGE_TYPE));
+
+        this.kinesisConsumer = new KinesisConsumer(nodeId, this, objectMapper, kinesisCallback(input),
+                                                   streamName, awsMessageType, batchSize, awsRequest);
 
         LOG.debug("Starting Kinesis reader thread for input [{}/{}]", input.getName(), input.getId());
         executor.submit(this.kinesisConsumer);
     }
 
+    /**
+     * If specified, verify that the specified endpoint is a valid URI.
+     *
+     * @param endpoint     The endpoint URI.
+     * @param endpointName The name of the endpoint. This will be included in the exception message.
+     * @throws MisfireException A {@link MisfireException} will be thrown if the URI is invalid.
+     */
+    private static void validateEndpoint(String endpoint, String endpointName) throws MisfireException {
+        if (StringUtils.isNotEmpty(endpoint)) {
```

Comment:
NB: Write a JUnit test to validate the endpoint , remove private block.

Reply:
Great point. Will do.

Reply:
Unit test added in https://github.com/Graylog2/graylog-plugin-integrations/pull/279/commits/8f341e6ac2abfda4d666124a1cce0a2798f18fce